src/parser/mod.rs

```diff
@@ -1174,6 +1175,14 @@ impl<'a> Parser<'a> {
                     ),
                 })
             }
+            Token::ExclamationMark if !self.dialect.supports_factorial_operator() => {
```

Comment:
```suggestion
            Token::ExclamationMark if self.dialect.supports_bang_not_operator() => {
```
See comment a bit lower for rationale to have a dedicated api for this behavior

Reply:
Similarly, this comment looks unadressed? The new feature ideally has its own `dialect` method. Otherwise it can get complicated to try to infer when a feature should take precedence

Reply:
The code has been re-pushed.